### PR TITLE
Update nginx-ingress-controller to 0.26.1

### PR DESCRIPTION
**Pull request by xrstf**
**What this PR does / why we need it**:
This has a fix for the issue plaguing our offline cluster. It was a smooth update on dev.

This PR also adds the `allowPrivilegeEscalation` field to the security context, which was set since 0.22 by https://github.com/kubernetes/ingress-nginx/pull/3342 but we never noticed because we don't have policies forbidding this behaviour by default.

The memory limit was increased because the release notes for 0.26 mentioned higher resource usage while connections are drained.

**Does this PR introduce a user-facing change?**:
```release-note
Update nginx-ingress-controller to 0.26.1
```

### CODE REVIEW

File: config/nginx-ingress-controller/values.yaml [hunk 1]
```diff
@@ -4,16 +4,18 @@ nginx:
   replicas: 3
   image:
     repository: quay.io/kubernetes-ingress-controller/nginx-ingress-controller
-    tag: 0.25.1
+    tag: 0.26.0
   config: {}
+  extraArgs:
+  - '--default-ssl-certificate=default/kubermatic-tls-certificates'
```
Comment (kron4eg) [inline config/nginx-ingress-controller/values.yaml:10]: What’s the reason behind this move? It used to be a strictly defined integration point between ingress controller and certmanager and now it became a possible to be overwritten and forgotten configuration option.

Comment (xrstf) [reply config/nginx-ingress-controller/values.yaml:10]: It felt weird to read some Kubermatic identifier within the nginx template, and a good opportunity to add the extraArgs stuff. But I don't feel strongly about it.

File: config/nginx-ingress-controller/Chart.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 name: nginx-ingress-controller
-version: 1.0.2
-appVersion: 0.25.1
+version: 1.0.3
+appVersion: 0.26.0
```
Comment (kron4eg) [inline config/nginx-ingress-controller/Chart.yaml]: Bugfix 0.26.1 is available

Comment (xrstf) [reply config/nginx-ingress-controller/Chart.yaml]: :trollface:

File: config/nginx-ingress-controller/values.yaml (hunk 1 above)
Comment (xrstf) [reply config/nginx-ingress-controller/values.yaml:10]: So.... keep it or remove it?

**Review by kron4eg — APPROVED** [review]
/lgtm

**Pull request merged**